tests/write_operations_test.go

```diff
@@ -0,0 +1,813 @@
+		fs := s.getFS(tbl)
+		s.writeParquet(fs, deleteFilePath, deleteData)
+		
+		// Create DataFile object for the delete file
+		deleteFileBuilder, err := iceberg.NewDataFileBuilder(
+			*iceberg.UnpartitionedSpec,
+			iceberg.EntryContentPosDeletes,
+			deleteFilePath,
+			iceberg.ParquetFile,
+			nil, // no partition data for unpartitioned
+			deleteData.NumRows(),
+			1024, // approximate file size
+		)
+		s.Require().NoError(err)
+		
+		deleteFile := deleteFileBuilder.Build()
+		
+		// Verify the delete file was created with correct properties
+		s.Equal(iceberg.EntryContentPosDeletes, deleteFile.ContentType())
+		s.Equal(deleteFilePath, deleteFile.FilePath())
+		s.Equal(int64(1), deleteFile.Count()) // One delete entry
+		
+		s.T().Log("Successfully created position delete file")
+	})
+	
+	s.Run("ApplyPositionDeletes", func() {
+		// Test reading with position deletes applied
+		ident := table.Identifier{"default", "position_delete_apply_test"}
+		tbl, originalFiles := s.createTableWithData(ident, 1)
+		
+		// The current iceberg-go scanner should be able to read and apply position deletes
+		// when they are properly associated with data files in manifests
+		
+		// For now, we'll test the delete file reading functionality
+		mem := memory.DefaultAllocator
+		posDeleteSchema := arrow.NewSchema([]arrow.Field{
+			{Name: "file_path", Type: arrow.BinaryTypes.String, Nullable: false},
+			{Name: "pos", Type: arrow.PrimitiveTypes.Int32, Nullable: false},
+		}, nil)
+		
+		// Create delete data
+		deleteData, err := array.TableFromJSON(mem, posDeleteSchema, []string{
+			fmt.Sprintf(`[
+				{"file_path": "%s", "pos": 0},
+				{"file_path": "%s", "pos": 2}
+			]`, originalFiles[0], originalFiles[0]),
+		})
+		s.Require().NoError(err)
+		defer deleteData.Release()
+		
+		// Write the position delete file
+		deleteFilePath := fmt.Sprintf("%s/deletes/pos_deletes_apply.parquet", s.location)
+		fs := s.getFS(tbl)
+		s.writeParquet(fs, deleteFilePath, deleteData)
+		
+		// Create DataFile object for the delete file
+		deleteFileBuilder, err := iceberg.NewDataFileBuilder(
+			*iceberg.UnpartitionedSpec,
+			iceberg.EntryContentPosDeletes,
+			deleteFilePath,
+			iceberg.ParquetFile,
+			nil,
+			deleteData.NumRows(),
+			1024,
+		)
+		s.Require().NoError(err)
+		
+		deleteFile := deleteFileBuilder.Build()
+		
+		// Test that we can read the delete file and verify its structure
+		// This demonstrates the delete file format is correct
+		s.Equal(int64(2), deleteFile.Count()) // Two delete entries
+		s.Equal(iceberg.EntryContentPosDeletes, deleteFile.ContentType())
+		
+		// NOTE: Full integration with table scanning would require updating
+		// manifests to include the delete files, which is a more complex operation
+		// that would need transaction-level support for delete file management
+		
+		s.T().Log("Position delete file structure verified - integration with table scanning requires manifest updates")
```

Comment:
s.T().Skip()

Reply:
iiuc this is the end of the test .. u mean to skip this test 

```
s.Run("ApplyPositionDeletes", func() {
```
?